File: backend/test/java/de/griefed/serverpackcreator/utilities/common/BooleanUtilitiesTest.java
package de.griefed.serverpackcreator.utilities.common;

import org.junit.jupiter.api.Assertions;
import org.junit.jupiter.api.Test;

public class BooleanUtilitiesTest {

  BooleanUtilities booleanUtilities;

  BooleanUtilitiesTest() {
    booleanUtilities = new BooleanUtilities();
  }

  @Test
  void convertToBooleanTestTrue() {
    Assertions.assertTrue(booleanUtilities.convert("True"));
    Assertions.assertTrue(booleanUtilities.convert("true"));
    Assertions.assertTrue(booleanUtilities.convert("1"));
    Assertions.assertTrue(booleanUtilities.convert("Yes"));
    Assertions.assertTrue(booleanUtilities.convert("yes"));
    Assertions.assertTrue(booleanUtilities.convert("Y"));
    Assertions.assertTrue(booleanUtilities.convert("y"));
  }

  @Test
  void convertToBooleanTestFalse() {
    Assertions.assertFalse(booleanUtilities.convert("False"));
    Assertions.assertFalse(booleanUtilities.convert("false"));
    Assertions.assertFalse(booleanUtilities.convert("0"));
    Assertions.assertFalse(booleanUtilities.convert("No"));
    Assertions.assertFalse(booleanUtilities.convert("no"));
    Assertions.assertFalse(booleanUtilities.convert("N"));
    Assertions.assertFalse(booleanUtilities.convert("n"));
  }
}

File: serverpackcreator-api/src/test/kotlin/de/griefed/serverpackcreator/api/common/BooleanUtilitiesTest.kt
package de.griefed.serverpackcreator.api.common

import de.griefed.serverpackcreator.api.utilities.common.BooleanUtilities
import org.junit.jupiter.api.Assertions
import org.junit.jupiter.api.Test

class BooleanUtilitiesTest internal constructor() {

    @Test
    fun convertToBooleanTestTrue() {
        Assertions.assertTrue(BooleanUtilities.convert("True"))
        Assertions.assertTrue(BooleanUtilities.convert("true"))
        Assertions.assertTrue(BooleanUtilities.convert("1"))
        Assertions.assertTrue(BooleanUtilities.convert("Yes"))
        Assertions.assertTrue(BooleanUtilities.convert("yes"))
        Assertions.assertTrue(BooleanUtilities.convert("Y"))
        Assertions.assertTrue(BooleanUtilities.convert("y"))
    }

    @Test
    fun convertToBooleanTestFalse() {
        Assertions.assertFalse(BooleanUtilities.convert("False"))
        Assertions.assertFalse(BooleanUtilities.convert("false"))
        Assertions.assertFalse(BooleanUtilities.convert("0"))
        Assertions.assertFalse(BooleanUtilities.convert("No"))
        Assertions.assertFalse(BooleanUtilities.convert("no"))
        Assertions.assertFalse(BooleanUtilities.convert("N"))
        Assertions.assertFalse(BooleanUtilities.convert("n"))
    }
}